saleor/graphql/product/types/products.py

```diff
@@ -43,7 +43,6 @@
 from ...utils import get_database_id, reporting_period_to_date
 from ..enums import OrderDirection, ProductOrderField
 from ..filters import AttributeFilterInput
-from ..resolvers import resolve_attributes
```

Comment:
Missed this part. Looks like the import is being used. Adding it back results in circular imports. And using lazy imports in the resolver file gives this error.
```
ERROR saleor.graphql.views Exception information: [PID:7088:Thread-1]
Traceback (most recent call last):
  File "C:\Users\khali\.virtualenvs\ksaleor-oqf0lJ8X\lib\site-packages\promise\promise.py", line 487, in _resolve_from_executor
    executor(resolve, reject)
  File "C:\Users\khali\.virtualenvs\ksaleor-oqf0lJ8X\lib\site-packages\promise\promise.py", line 754, in executor
    return resolve(f(*args, **kwargs))
  File "C:\Users\khali\.virtualenvs\ksaleor-oqf0lJ8X\lib\site-packages\graphql\execution\middleware.py", line 75, in make_it_promise
    return next(*args, **kwargs)
  File "D:\Workspace\I-Novanet\Manjoo\Backend\ksaleor\saleor\graphql\core\fields.py", line 144, in connection_resolver
    iterable = resolver(root, info, **args)
  File "D:\Workspace\I-Novanet\Manjoo\Backend\ksaleor\saleor\graphql\product\schema.py", line 247, in resolve_attributes
    return resolve_attributes(info, **kwargs)
  File "D:\Workspace\I-Novanet\Manjoo\Backend\ksaleor\saleor\graphql\product\resolvers.py", line 47, in resolve_attributes
    category = graphene.Node.get_node_from_global_id(info, in_category, "Category")
  File "C:\Users\khali\.virtualenvs\ksaleor-oqf0lJ8X\lib\site-packages\graphene\relay\node.py", line 102, in get_node_from_global_id
    only_type._meta.name
AttributeError: 'str' object has no attribute '_meta'
```
Any clue how to work around this?

Reply:
You could use `from_global_id_strict_type` which returns a `pk` (as str, don't need to cast it). You can provides it with the keyword argument `only_type` which supports lazy types. Then you can manually get the node through `category = models.Category(pk=pk)` but it will raise `ObjectDoesNotExist`, which thus needs to be wrapped into a try-catch.

But that would make the code a bit dirty. You could add a django-like shortcut: `get_or_error(field, model or queryset, **kwargs)` function into `saleor.graphql.core.utils`. Note: `field` would be used for `raise ValidationError({field: "message here"})`.

I think it would be better as it give us more control over how to resolve the node into a model instance (prefetch, etc.).

So you can do this:
```python
category = get_node_or_error("in_category", models.Category, pk=pk)
```

Which in other places would be useful for things like:
```python
qs = MyRelatedModel.objects.only("pk").prefetch("pizzas")
obj = get_node_or_error("model_id", qs, user_id=user_id, checkout_id=checkout_id)
```

Reply:
Actually, we do have one in the core mutation `get_node_or_error`. You can invoke it through `BaseMutation.get_node_or_error`. For now it should be fine, but we should extend it a bit and extract it to a simple function instead of `classmethod` but that's for another PR.

Reply:
`BaseMutation.get_node_or_error` makes a call to `get_node_from_global_id`, passing `only_type` as a str. Using it produces the same error as before.

I tried adding a `get_node_or_error` in `graphql.core.utils`. Could you take a look at https://github.com/mirumee/saleor/blob/62b3f6ab29003ea1fa018c2b2ad1a81ae48ed4da/saleor/graphql/core/utils/__init__.py#L37
At this point, I have `only_type` as a str and `graphene_type` as a `<class 'graphene.utils.subclass_with_meta.SubclassWithMeta_Meta'>`, so the check fails and the validation error is raised. Is it fine to cast graphene_type to str here?

Not very familiar with the graphql code in saleor still.